### PR TITLE
Fix darwin build and bumup nixpkgs

### PR DESCRIPTION
**Pull request by GTrunSec**
I'm going to fix the PR of https://github.com/tweag/jupyterWith/pull/173 problem.

### CODE REVIEW

**Comment by GTrunSec** [comment]
@guaraqe  ready to review.

**Comment by guaraqe** [comment]
@teto Do you confirm this corresponds to what you expected from the PR?

**Comment by teto** [comment]
thanks a lot for looking into it ! super appreciated.

If I do `nix build .\#jupyterEnvironment --out-link result`
```
jupyterWith on  use-ihaskell-flake 
➜ nix build .\#jupyterEnvironment --out-link result
jupyterWith on  use-ihaskell-flake 
➜ result/bin/jupyter-lab
[I 2021-12-13 10:55:02.409 ServerApp] jupyterlab | extension was successfully linked.
[I 2021-12-13 10:55:02.428 LabApp] JupyterLab extension loaded from /nix/store/kdbj6hkdn8sc8agmd0jpwwf4byjwnb13-python3.9-jupyterlab-3.2.3/lib/python3.9/site-packages/jupyterlab
[I 2021-12-13 10:55:02.428 LabApp] JupyterLab application directory is /home/teto/jupyterWith/.jupyterlab
[E 2021-12-13 10:55:02.429 LabApp] JupyterLab application assets not found in "/home/teto/jupyterWith/.jupyterlab"
[E 2021-12-13 10:55:02.429 LabApp] Please run `jupyter lab build` or use a different app directory
[I 2021-12-13 10:55:02.431 ServerApp] jupyterlab | extension was successfully loaded.
[I 2021-12-13 10:55:02.431 ServerApp] Serving notebooks from local directory: /home/teto/jupyterWith
[I 2021-12-13 10:55:02.431 ServerApp] Jupyter Server 1.11.2 is running at:
[I 2021-12-13 10:55:02.431 ServerApp] http://localhost:8888/lab?token=c6622280e0d5329b37bc7251e7d609f8bbce904036b611a5
[I 2021-12-13 10:55:02.431 ServerApp]  or http://127.0.0.1:8888/lab?token=c6622280e0d5329b37bc7251e7d609f8bbce904036b611a5
[I 2021-12-13 10:55:02.431 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2021-12-13 10:55:02.516 ServerApp] 
    
    To access the server, open this file in a browser:
        file:///home/teto/.local/share/jupyter/runtime/jpserver-408549-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/lab?token=c6622280e0d5329b37bc7251e7d609f8bbce904036b611a5
     or http://127.0.0.1:8888/lab?token=c6622280e0d5329b37bc7251e7d609f8bbce904036b611a5
     
```
and the webpage content is
```
JupyterLab Error
JupyterLab application assets not found in "/home/teto/jupyterWith/.jupyterlab"
Please run `jupyter lab build` or use a different app directory
```
so I think either we miss a flag or we should remove a flag from the ihaskell flake. We should fix this before merge other than that it's good. We can refine later (use same python in ihaskell and jupyterWith for instance, add ihaskell cache to jupyterWith flake etc) but it will be easier once this gets in.

**Comment by GTrunSec** [comment]
>so I think either we miss a flag or we should remove a flag from the ihaskell flake. We should fix this 

first, you can remote the local flag and build the `jupterlab` directory in nix store with the impure of nix-shell. or give the local directory a shellhook to run the jupyter lab build automatically.

File: flake.nix [hunk 1]
```diff
@@ -2,48 +2,90 @@
...
+            packages = p: with p; [ vector aeson ];
+            extraIHaskellFlags = "--codemirror Haskell"; # for jupyterlab syntax highlighting
+            haskellPackages = pkgs.haskellPackages;
+          };
+
+        in
+        {
+          jupyterEnvironment = pkgs.jupyterWith.jupyterlabWith {
+              kernels = [ iPython iHaskell ];
+              directory = "./.jupyterlab";
```
Comment (teto) [inline flake.nix:53]: removing this line makes it possible to find back master's behavior. Remove this and then it's ready for merge I think

**Review by teto — APPROVED** [review]

**Comment by guaraqe** [comment]
@teto This means we can close the other PR, right?

**Comment by teto** [comment]
@guaraqe yes but apparently it was closed automatically. Thanks for merging this. I have a custom ihaskell I wanted to try with jupyterWith and this makes it so much easier !

**Pull request merged**